backend/effects/builtin/toggle-command.js

```diff
@@ -0,0 +1,112 @@
+                <ui-select-match placeholder="Select or search for a command... ">{{$select.selected.trigger}}</ui-select-match>
+                <ui-select-choices repeat="command.id as command in systemCommands | filter: { trigger: $select.search }" style="position:relative;">
+                    <div ng-bind-html="command.trigger | highlight: $select.search"></div>
+                </ui-select-choices>
+            </ui-select>
+        </eos-container>
+
+        <eos-container ng-show="effect.commandType === 'custom'" header="Custom Commands" pad-top="true">
+            <ui-select ng-model="effect.commandId" theme="bootstrap" ng-click="updateSelectedCommand(customCommands)">
+                <ui-select-match placeholder="Select or search for a command... ">{{$select.selected.trigger}}</ui-select-match>
+                <ui-select-choices repeat="command.id as command in customCommands | filter: { trigger: $select.search }" style="position:relative;">
+                    <div ng-bind-html="command.trigger | highlight: $select.search"></div>
+                </ui-select-choices>
+            </ui-select>
+        </eos-container>
+
+        <eos-container header="Toggle Action" pad-top="true">
+            <dropdown-select options="toggleOptions" selected="effect.toggleType"></dropdown-select>
+        </eos-container>
+    `,
+    optionsController: ($scope, commandsService) => {
+        $scope.systemCommands = commandsService.getSystemCommands();
+        $scope.customCommands = commandsService.getCustomCommands();
+
+        $scope.updateSelectedCommand = commands => {
+            if ($scope.effect.commandId) {
+                for (const command of commands) {
+                    if (command.id === $scope.effect.commandId) {
+                        $scope.effect.command = command;
+                    }
+                }
+            }
+        };
+
+        $scope.toggleOptions = {
+            disable: "Deactivate",
+            enable: "Activate"
+        };
+
+        if ($scope.effect.toggleType == null) {
+            $scope.effect.toggleType = "disable";
+        }
+    },
+    optionsValidator: effect => {
+        let errors = [];
+        if (effect.commandId == null) {
+            errors.push("Please select a command.");
+        }
+        return errors;
+    },
+    onTriggerEvent: async event => {
+        const { effect } = event;
+        let command = effect.command;
+        command.active = effect.toggleType === "enable";
+
+        if (command.id == null || command.id === "") return;
+
+        if (command.type === "system") {
+            commandManager.saveSystemCommandOverride(command);
+        }
+
+        if (command.type === "custom") {
+            command.lastEditAt = moment().format();
+
+            commandManager.saveCustomCommand(command, event.trigger.metadata.username, false);
+        }
+
+        commandAccess.triggerUiRefresh();
```

Comment:
We should be able to get the most current command/syscommand using the `commandId` and `commandType` effect properties instead of caching the entire command in the effect which could lead to stale data (see comment above).

Here's how we could accomplish that:
```js
const { commandId, commandType, toggleType } = effect;

if (commandType === "system") {
    const systemCommand = commandManager
        .getAllSystemCommandDefinitions().find(c => c.id === commandId);

    if (systemCommand == null) {
        // command doesnt exist anymore
        return true;
    }

    systemCommand.active = toggleType === "toggle" ? !systemCommand.active : toggleType === "enable";

    commandManager.saveSystemCommandOverride(systemCommand);

    frontendCommunicator.send("systemCommandsUpdated");
} else if (commandType === "custom") {
    const customCommand = commandManager.getCustomCommandById(commandId);

    if (customCommand == null) {
        // command doesnt exist anymore
        return true;
    }

    customCommand.active = toggleType === "toggle" ? !customCommand.active : toggleType === "enable";

    commandManager.saveCustomCommand(customCommand, "System", false);
}
```

Reply:
Ahhh this is perfect! Thank you so much for explaining all this, and for showing me the correct solution. It's really helpful and educational. I updated the PR according to all of the feedback, including the toggle state. 😄